bundles/binding/org.openhab.binding.cardio2e/src/main/java/org/openhab/binding/cardio2e/internal/Cardio2eTransactionParser.java

```diff
@@ -0,0 +1,463 @@
+/**
+ * Copyright (c) 2010-2015, openHAB.org and others.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.cardio2e.internal;
+
+import java.util.Calendar;
+import org.openhab.binding.cardio2e.internal.code.Cardio2eCurtainTransaction;
+import org.openhab.binding.cardio2e.internal.code.Cardio2eDateTime;
+import org.openhab.binding.cardio2e.internal.code.Cardio2eDateTimeTransaction;
+import org.openhab.binding.cardio2e.internal.code.Cardio2eHvacControlTransaction;
+import org.openhab.binding.cardio2e.internal.code.Cardio2eHvacTemperatureTransaction;
+import org.openhab.binding.cardio2e.internal.code.Cardio2eLightingTransaction;
+import org.openhab.binding.cardio2e.internal.code.Cardio2eRelayTransaction;
+import org.openhab.binding.cardio2e.internal.code.Cardio2eScenarioTransaction;
+import org.openhab.binding.cardio2e.internal.code.Cardio2eSecurityTransaction;
+import org.openhab.binding.cardio2e.internal.code.Cardio2eTransaction;
+import org.openhab.binding.cardio2e.internal.code.Cardio2eTransactionTypes;
+import org.openhab.binding.cardio2e.internal.code.Cardio2eZonesBypassTransaction;
+import org.openhab.binding.cardio2e.internal.code.Cardio2eZonesTransaction;
+import org.openhab.core.items.Item;
+import org.openhab.core.library.types.DateTimeType;
+import org.openhab.core.library.types.DecimalType;
+import org.openhab.core.library.types.IncreaseDecreaseType;
+import org.openhab.core.library.types.OnOffType;
+import org.openhab.core.library.types.OpenClosedType;
+import org.openhab.core.library.types.PercentType;
+import org.openhab.core.library.types.StopMoveType;
+import org.openhab.core.library.types.UpDownType;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.State;
+
+/**
+ * This class is responsible for parsing the Cardio 2é transactions.
+ * 
+ * @author Manuel Alberto Guerrero Díaz
+ * @since 1.7.0
+ */
+class Cardio2eTransactionParser {
+
+	protected static boolean filterUnnecessaryCommand;
+	protected static boolean filterUnnecessaryReverseModeUpdate;
+
+	
+	static void commandToTransaction (Command command, Cardio2eTransaction transaction) {
+		commandToTransaction (command, transaction, null);
+	}
+	
+	static void commandToTransaction (Command command, Cardio2eTransaction transaction, String securityCode) {
+		transaction.mustBeSent = false;
+		switch (transaction.getObjectType()) {
+		case LIGHTING:
+			Cardio2eLightingTransaction lightingTransaction = (Cardio2eLightingTransaction) transaction;
+			byte initialLightIntensityValue = lightingTransaction.getLightIntensity();
+			switch (command.getClass().getSimpleName()) {
+			case "OnOffType":
```

Comment:
Please use constants for OnOffType, IncreaseDecreaseType, PercentType and so on. You have those fixed strings many times in your code.